engine/src/exec-array.cpp

```diff
@@ -606,12 +606,23 @@ void MCArraysDoIntersect(MCExecContext& ctxt, MCArrayRef p_dst_array, MCArrayRef
 	MCNameRef t_key;
 	MCValueRef t_src_value;
     MCValueRef t_dst_value;
+    MCArrayRef p_orig_array;
```

Comment:
I'd suggest making this variable t_orig_dst_array - our coding convention is that temporaries should begin with "t_".